core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java

```diff
@@ -42,6 +46,10 @@
     private final ClientDecorationBuilder decoration = ClientDecoration.builder();
     private final HttpHeadersBuilder httpHeaders = HttpHeaders.builder();
 
+    private boolean basicAuthSet;
+    private boolean oAuth1aSet;
+    private boolean oAuth2Set;
```

Comment:
I think it's possible that a user can call `auth()` method multiple times with different tokens. For example, a user could set the default `BasicToken` and then override it as it reads some additional configuration file that specifies a different token.

Reply:
If the client sets tokens for the same authentication, it does not throw an exception.
It's throwing an exception only when two different tokens for two different authentications such as basic and OAuth 2.0 are set.
Do you think it's better not to throw an exception in that case?

Reply:
Yeah, that was what I meant. I think a user might want to override the authentication mechanism. For example, one module could return a pre-configured `ClientBuilder`, allowing a user to override some of its settings. Do you have any concern on this idea?

Reply:
Just worried about a case where a user misconfigured, but it seems OK. Let me just remove this check to simplify, thanks!

Reply:
On my second thought, I think finding the cause is not as easy when accidentally two different auths are used if we don't throw an exception because:
- the client should actually send a request
- it's hard to get to know right away that the authentication header is the problem from the response content.
Meanwhile, fixing pre-configure `ClientBuilder` is really easy.
So can I keep it as it is?

Reply:
I'm not sure if it's a good idea yet because:

- Our other setters do not prohibit setting more than once. Why should this method be special?
- A server will give a client 403 forbidden anyway when misconfigured (including bad password).

Reply:
Removed checking auth setting. 😄

Reply:
Thanks a lot!